products/facilities/regional-offices/engineering/testing-qa.md

```diff
@@ -1 +1,90 @@
+    1. For example: since we wanted to make sure multiple banner types can appear on a page, we build a CMS node of a VBA facility with a banner alert and take a segment of the generated JSON from Drupal. We repeat this for all banner types. Then we use renderHTML to build the template pieces and 'expect' that each banner type appears when the mock data is present for it. Since banner templates are built on `<va-banner>` components, we are only able to test props of the Design System Components.
+4. There is a limitation to testing via renderHTML because it does not pull in or load react components and it does not allow us to test deep in the shadow DOM. So for all Design System Components (which we rely on) we expect that the Design System Team has tested for certain accessibility  features. Our manual checks also verify the Design System Components are acting as expected.
+5. To facilitate processing the drupal content into templated pages, we often use liquid filters. Whenever we create a liquid filter we add tests to the CI pipeline for that filter. We always cover null conditions and valid object types (because all data from Drupal comes as objects). You will see below the list of liquid filters added and the tests created.
+
+## Regression Testing
+Regression testing is built into the automated testing and the CI will not succeed in building and deploying the content if it does not comply with the automated testing. However, if CMS content is changed and is not congruent with the structure of the GraphQL the CI will not succeed. Testing should ensure for all potential options that an editor is presented with for the particular component in the VBA Regional Benefit Office pages.
+
+## Testing currently existing as of 4/19/24
+### Liquid filters
+- processVbaServices
+  - generates Facility service or Service Region or both with specific fieldVbaTypeOfCare and tests that services with correct types of care are grouped into correct object locations.
+- processfieldCcCantFindBenefits
+  - checks for null, valid object
+- processCentralizedBenefitsHotline
+  - checks for null, valid object
+- processCentralizedUpdatesVBA
+  - checks for null, valid object
+
+### renderHTML
+#### Banner
+- src/site/layouts/tests/vba/template/vba_banner_test.unit.spec.js
+  - Tested with renderHTML
+    - dismissible info banner
+    - dismissible warning banner
+    - non-dissmisible warning banner
+
+#### Accordions
+- src/site/tests/vba/vba.unit.spec.js
+  - tests accordions get generated from mock of VBA. Tests that accordions of specific types are created.
+    - (suggestion: A more comprehensive service location generator should be created which can test all possible values for field office visits, facility main phone number presence, etc).
```

Comment:
We could phrase this as: 

    - CONSIDER: Our team is considering creating a more comprehensive service location generator which can test all possible values for field office visits, facility main phone number presence, etc.  

If we mean that, let's cut a stubby ticket to track it? Or should that wait til after we talk through this in Staging Review maybe?

Reply:
This is the thing you were talking about generating a script to do (generate the content on the CMS). That then we would just download the one complete VBA facility and put it in place of the mock facility.

Reply:
Updated to:
```
- CONSIDER: Our team is considering creating a more comprehensive service location generator which can generate all possible values for field office visits, facility main phone number presence, etc. This will allow us to process all possible values in the service location accordions.
```